ui/src/routes/devices.$id.settings.access._index.tsx

```diff
@@ -187,9 +193,17 @@ export default function SettingsAccessIndexRoute() {
     setTlsKey(value);
   };
 
+  const handleTlsEnforceChange = (value: boolean) => {
+    setTlsEnforce(value);
+    // For "disabled" and "self-signed" modes, immediately apply the settings
+    if (tlsMode !== "custom") {
```

Comment:
Would it be better to only immediately apply setting to `"disabled"`.

Also we should **not** apply the force option being turned on until an explicit Save click (like we do for many of the Network settings), so that we don't immediately lock someone out of their device because of an errant click?

Reply:
I have made it so that Enforce TLS can only be enabled after acknowledging a pop-up and the user applies TLS settings (in "custom" / "self-signed") mode.